### PR TITLE
RUN protocol governance to start PSM, runStake, etc.

### LINKED ISSUE #3788: RUNstake: borrow RUN against staked BLD
## What is the Problem Being Solved?

per [RUNStake User Stories](https://docs.google.com/document/d/19xP4AjLDdHvlGYFyC_Yq3Opp7oR60rRc-9UKAyb26a4/edit#heading=h.ekia0c1wmi3q) section by @rowgraus and co 5 Apr:

As a BLD staker with no RUNstake position, I need to be able to open a position by placing a lien on my staked BLD and receiving RUN
 - Process should feel familiar to vaults
 - Subject to RUN per BLD limits
 - Subject to asset level debt limit

As a BLD staker with an existing RUNStake position, I need to be able to manage it; that is: adjust my RUN debt and BLD lien within the bounds of the contract.
 - Process should feel familiar to vaults

As a BLD staker with an existing RUNStake position, I need to be able to close it entirely
 - May be combined with/handled by the ‘manage’ use case above

## Description of the Design

Like vaultFactory, the runStake contract provides loans and charges interest and minting fees; differences include:
  1. The loan is secured by a lien on staked BLD, not by transferring collateral.
  2. There is no liquidation. Instead, the staked BLD generates a stream of RUN rewards which can be used to pay down the debt. (See #4244 for more on that part.)

Design notes in [runStake/README.md](https://github.com/Agoric/agoric-sdk/blob/master/packages/run-protocol/src/runStake/README.md) include a sequence diagram of interactions between a user/dapp, walletBridge, runStake (and attestation), and Cosmos SDK.

We use a semi-fungible `copyBag` token to represent lien attestations: the items is the address where BLD is staked and the quantity is the lien (delta).

 - [x] add `getLiened()` to [attMaker](https://github.com/Agoric/agoric-sdk/blob/dc-run-loc/packages/zoe/src/contracts/attestation/attestation.js#L97) so the wallet can get it before taking out a loan.

related work:

 - #3277
 - #4165
 - #4245
 - #4244
 - #4263
 - #3991 

## Security Considerations

 - line of credit involves use of the RUN mint
 - BLD liens merit care

## Test Plan

 - [x] [RUN LoC Test Cases \- Google Sheets](https://docs.google.com/spreadsheets/d/1XC754C9d6KyEuYkMjrSWNi5xDSvnL6-oc8DQw26vtn4/edit#gid=0)
   - [x] payback
 - [x] use governance for minting ratio
 - [x] charge interest
 - [x] Attestation tokens control liens
 - [x] bootstrap provides `attestationMaker` to each `home.wallet` as in https://github.com/Agoric/agoric-sdk/issues/3788#issuecomment-994100862
 - [ ] notifier for wallet (#4263)
 - [x] `local-chain` governance testing documentation
 
nice to have:
 - [ ] generate sequence diagram from a test and check against runstake/README.md

### PR DESCRIPTION
**Pull request by dckc**
obsoletes #4953
closes: #3788
refs: #5010
  - would close it but for governance, which should have its own ticket, but does not, yet

## Description

Break the default run-protocol governance proposal used in sim-chain / loadgen down into parts:

- the economic committee
- runStake
- AMM, vaults, reward distributor
- PSM

### Security Considerations

The manifests / permits are an enforced security policy. (_We have visualizations, but maintenance is a little iffy._)

### Documentation Considerations

#5062 is a plan to launch the RUN protocol contracts on devnet by vote of BLD holders.

@rowgraus I'd like to brainstorm a bit about how we explain this stuff to the voters. [devnet: proposal to deploy RUNstake passed](https://commonwealth.im/agoric/discussion/4295-devnet-proposal-to-deploy-runstake-passed) was an post-facto explanation of an earlier episode to our commonwealth forum.

We're inviting not just our devnet validators but all our keplr users to participate in this preview of governance: https://github.com/Agoric/documentation/issues/668

### Testing Considerations

I'd like someone else to reproduce my results; I'm working with @samsiegart .

Instructions (sketch) so far:

branch:  dc-boot-psm-runStake

start local chain:

```
~/projects/agoric/agoric-sdk/packages/cosmic-swingset
12:46 connolly@jambox$ make scenario2-setup scenario2-run-chain >,chain.log 2>&1 &
```

start local solo:
```
~/projects/agoric/agoric-sdk/packages/cosmic-swingset
12:46 connolly@jambox$ make scenario2-run-client >,client.log 2>&1 &
```

deploy contracts; make proposals:

```
cd packages/run-protocol
yarn deploy
```

then...
(I used a different terminal...)

```
cd ../cosmic-swingset
make start-committee VOTE_PROPOSAL=1 # salt to taste
make start-runStake VOTE_PROPOSAL=2
make gov-q
```

### CODE REVIEW

File: packages/deploy-script-support/src/installInPieces.js [hunk 1]
```diff
@@ -1,16 +1,44 @@
+/* global globalThis */
 // @ts-check
 import { E } from '@endo/far';
 import { ZipReader } from '@endo/zip';
 import { encodeBase64, decodeBase64 } from '@endo/base64';
+import { importBundle } from '@endo/import-bundle';
```
Comment (michaelfig) [inline packages/deploy-script-support/src/installInPieces.js]: As lint says, it's necessary to add `@endo/import-bundle` to `packages/deploy-script-support/package.json`.dependencies.

File: packages/run-protocol/scripts/init-core.js [hunk 2]
```diff
@@ -1,68 +1,169 @@
...
+const mapValues = (obj, f) =>
+  // @ts-ignore entries() loses the K type
+  harden(fromEntries(entries(obj).map(([p, v]) => [p, f(v)])));
+
+const committeeProposalBuilder = async ({ publishRef, install }) => {
+  const { ROLE = 'chain' } = process.env;
+
+  // preload ERTP, marshal, store, etc.
+  const [mod0, bundle0] = installKeyGroups.econCommittee.binaryVoteCounter;
+  await install(mod0, bundle0, { persist: true });
```
Comment (michaelfig) [inline packages/run-protocol/scripts/init-core.js]: Would this create the first of two separate Zoe installations?  Should the below `publishGroup` avoid doing an install for `group === 'econCommittee' && mod === 'binaryVoteCounter'`?

File: packages/run-protocol/src/core-proposal.js [hunk 3]
```diff
@@ -201,3 +275,53 @@ export const getManifestForRunProtocol = (
...
+      runStake: restoreRef(installKeys.runStake),
+    },
+  };
+};
+
+export const getManifestForPSM = ({ restoreRef }, { installKeys }) => {
+  return {
+    manifest: PSM_MANIFEST,
+    installations: {
+      runStake: restoreRef(installKeys.runStake),
```
Comment (michaelfig) [inline packages/run-protocol/src/core-proposal.js:337]: Looks like this shouldn't be `runStake`?

**Review by michaelfig — APPROVED** [review]
Looks pretty good!  Please don't merge until the `psm/installations/runStake` thing is sorted.

File: packages/deploy-script-support/src/installInPieces.js [hunk 4]
```diff
@@ -1,16 +1,44 @@
...
+};
 
 export const installInPieces = async (
   bundle,
   bundler,
-  { maxBytesInFlight = 800_000, log = console.log } = {},
+  { maxBytesInFlight = 800_000, log = console.log, persist = false } = {},
 ) => {
+  // Detect evaluation errors early.
+  await importBundle(bundle, { endowments });
```
Comment (dckc) [inline packages/deploy-script-support/src/installInPieces.js]: @michaelfig , this seems to work for me, but when@samsiegart does `yarn deploy:proposal`, he gets the following error. Help? Clues?

```
Bundling bundle maker /home/samsiegart/agoric-sdk/packages/deploy-script-support/src/endo-pieces-contract.js
agoric: (Error#1)
Error#1: VatData unavailable

  at unvailable (.../vat-data/src/index.js:28:29)
  at eval (.../run-protocol/src/vaultFactory/liquidateMinimum.js:105:31)
  at async installInPieces (packages/deploy-script-support/src/installInPieces.js:28:3)
  at async publishRef (packages/deploy-script-support/src/writeCoreProposal.js:77:19)
  at async Promise.all (index 2)
  at async Promise.all (index 2)
  at async Promise.all (index 1)
  at async Promise.all (index 1)

Error#1 ERROR_NOTE: Rejection from: (Error#2) : 24 . 8
Nested error under Error#1
  Error#2: Event: 20.6
  
    at writeCoreProposal (packages/deploy-script-support/src/writeCoreProposal.js:82:28)
    at default (packages/run-protocol/scripts/init-core.js:178:5)
    at async WebSocket.<anonymous> (packages/agoric-cli/src/deploy.js:274:13)
  
error Command failed with exit code 2.
```

File: packages/run-protocol/scripts/init-core.js (hunk 2 above)
Comment (dckc) [reply packages/run-protocol/scripts/init-core.js]: Yes, it makes a throw-away installation, but I didn't think it was worth fixing. Will do.

File: packages/deploy-script-support/src/installInPieces.js (hunk 4 above)
Comment (dckc) [reply packages/deploy-script-support/src/installInPieces.js]: this is a symptom of calling `importBundle` outside swingset. Let's not, for now.

**Comment by dckc** [comment]
@samsiegart when you can get `home.attMaker`, please approve this.

**Review by turadg — COMMENTED** [review]
No concerns from me.

**Comment by dckc** [comment]
missing lienBridge

**Comment by samsiegart** [comment]
I'm able to get `home.attMaker` on [22f08bc](https://github.com/Agoric/agoric-sdk/commit/22f08bcab784921513924a7164bca622d5fa438c)

**Review by samsiegart — APPROVED** [review]
```
command[4]
E(home.attMaker).getAccountState()
history[4]
{"bonded":{"brand":[Object Alleged: BLD brand]{},"value":0n},"currentTime":1650555791n,"liened":{"brand":[Object Alleged: BLD brand]{},"value":0n},"locked":{"brand":[Object Alleged: BLD brand]{},"value":0n},"total":{"brand":[Object Alleged: BLD brand]{},"value":13000000n},"unbonding":{"brand":[Object Alleged: BLD brand]{},"value":0n}}
```

File: packages/agoric-cli/src/deploy.js [hunk 5]
```diff
@@ -191,7 +191,7 @@ export default async function deployMain(progname, rawArgs, powers, opts) {
                 paths: [...paths, path.dirname(moduleFile)],
               });
             } catch (e) {
-              return path.resolve(...paths, fileName);
+              return path.resolve(path.dirname(moduleFile), ...paths, fileName);
```
Comment (mhofman) [inline packages/agoric-cli/src/deploy.js:194]: Why was `path.dirname(moduleFile)` added before `...paths` ?

Comment (mhofman) [reply packages/agoric-cli/src/deploy.js:194]: Nmv, this is correct and it's the `paths` for `require.resolve` above that isn't correct.

**Pull request merged**